src/error.rs

```diff
@@ -15,6 +15,12 @@ impl embedded_hal::digital::Error for MockError {
     }
 }
 
+impl embedded_can::Error for MockError {
```

Comment:
i suspect `MockError` is going to need to reflect the variants in [embedded_can::ErrorKind](https://docs.rs/embedded-can/latest/embedded_can/enum.ErrorKind.html) so mocks can deal with at least contention (`Bit`) and `CRC` errors... but we can probably expand on this later.